### ISSUE TITLE
Don't defer updating threadmark thread_id when moving posts

### ISSUE THREAD

**Issue by sidane** [Sidane/xenforo-threadmarks#30]
See #17.

When moving a threadmarked post to a different thread, it would be preferable to not defer the updating of a threadmark's `thread_id.

It takes a few seconds (or longer if larger forums are running deferred tasks via cron) for the threadmark to appear in the target thread. This is likely to lead to confusion among moderators/admins, they may think the threadmark has been lost during the move.

Investigate alternatives to deferring.

**Comment by Xon** [comment]
One solution may be to relax the join condition in Sidane_Threadmarks_Model_Post::preparePostJoinOptions to exclude the thread_id

```
LEFT JOIN threadmarks  ON
          (threadmarks.thread_id = post.thread_id && threadmarks.post_id = post.post_id)
```

It would require a post_id index on threadmarks since MySQL may not be able to use the index without the thread_id.

Then in the display code, if the thread_id from the threadmark doesn't match the thread; display a spinner on the presumption a deferred task is working on cleaning it up.

**Comment by Xon** [comment]
PR #40 should vastly improve this.

**Comment by sidane** [comment]
Yep, a definite improvement, thanks. 

All that remains in this issue is that the threadmarks menu isn't visible until the deferred task completes.

**Comment by Xon** [comment]
Fully resolved in #49
